### PR TITLE
[hotfix] Fix deleting Puffin files when cleaning orphan files

### PR DESCRIPTION
**Pull request by wangtaohz**
## Why are the changes needed?

The puffin files are in the `data/puffin` location which should also be protected when cleaning the data files.

## Brief change log

- should not delete Puffin files when cleaning orphan files

## How was this patch tested?

- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / **no**)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java [hunk 1]
```diff
@@ -324,12 +335,16 @@ public BaseTableMaintainer(UnkeyedTable unkeyedTable) {
 
     @Override
     public Set<String> orphanFileCleanNeedToExcludeFiles() {
-      return mergeSets(changeFiles, baseFiles, hiveFiles);
+      return ImmutableSet.<String>builder()
```
Comment (shidayang) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java:338]: Implement mergeSets using ImmutableSet?

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java:338]: OK, I will extract a method to make the code more concise.

Comment (wangtaohz) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/MixedTableMaintainer.java]: I change to using the `Sets#union`.

**Review by zhoujinsong — APPROVED** [review]
LGTM.

**Pull request merged**